hws/src/dr_remote.py

```diff
@@ -0,0 +1,113 @@
+#SPDX-License-Identifier: BSD-3-Clause
+#Copyright (c) 2024 NVIDIA CORPORATION. All rights reserved.
+
+import importlib
+import time
+import sys
+import os
+
+from src.dr_common import *
+from src.dr_db import _config_args
+
+SSH_OPTIONS = "-q -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null -oPubkeyAuthentication=no"
+
+def remote_cmd(cmd, host, user, password, time_out=60):
+    if _config_args.get("remote_dep_lib") == False:
+        globals()["pexpect"] = importlib.import_module("pexpect")
+        globals()["tempfile"] = importlib.import_module("tempfile")
+        _config_args["remote_dep_lib"] = True
+
+    f = tempfile.mktemp()
+    fout = open(f, 'w')
+
+    cmd = "ssh %s@%s %s \"%s\"" % (user, host, SSH_OPTIONS, cmd)
+    p = pexpect.spawnu(cmd, timeout=time_out)
+  
+    i = p.expect(['[Yy]es', 'YES', '[Pp]assword: '])
+    if i < 2:
+        p.sendline('YES')
+        i = p.expect(['[Yy]es', 'YES', '[Pp]assword: '])
+        if i < 2:
+            p.sendline('Y')
+            i = p.expect(['[Yy]es', 'YES', '[Pp]assword: '])
+            if i < 2:
+                print("Connection error")
+                sys.exit(1)
+
+    p.sendline(password)
+    p.logfile = fout
+    p.expect(pexpect.EOF)
+    p.close()
+    fout.close()
+
+    fin = open(f, 'r')
+    stdout = fin.read()
+    fin.close()
+
+    if p.exitstatus != 0:
+        raise Exception(stdout)
+
+    return stdout
+
+
+def build_options():
+    op = ""
+
+    if _config_args.get("device") != None:
+        op += "-d %s " % (_config_args.get("device"))
+
+    if _config_args.get("verbose") > 0:
+        op += "-%s " % (_config_args.get("verbose") * "v")
+
+    if _config_args.get("statistics") == True:
+        op += "-s "
+
+    if _config_args.get("parse_hw_resources") == False:
+        op += " --skip_parse"
+
+    return op
+
+
+def dr_connect_to_remote():
+    user_name = _config_args.get("user_name")
+    raw_dump_file = _config_args.get("file_path")
+    remote_ip = _config_args.get("remote_ip")
+    options = build_options()
+    remote_dump_file = "_%s_%s" % (str(time.time()), raw_dump_file)
```

Comment:
I'd switch the order of strings in the remote_dump_file: instead of _time_filename, make it _filename_time. This way when you do "ls" you will see them all together

Reply:
ack